subsys/usb/class/netusb/netusb.c

```diff
@@ -188,6 +196,8 @@ int try_write(u8_t ep, u8_t *data, u16_t len)
 			 */
 			if (tries--) {
 				SYS_LOG_WRN("Error: EAGAIN. Another try");
+				/* Wait for the transfer to complete */
+				k_sem_take(&netusb.bulk_in_sem, K_MSEC(100));
 				continue;
```

Comment:
Just to check, could the issue be solved by k_yield() (or even k_sleep() despite it does not look nice)? BTW try_write() would be replaced by usb_transfer(), there is a patch for ECM in PR #5983 .

Reply:
Ack, thanks.

re: k_yield() it might reduce the problem but it's hard to say.  USB is fast so you don't need to spend much time for the transfer to have completed - for example, if logging is on then the SYS_LOG_WRN() is enough.

re: transfers, I'll look into it.